sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala

```diff
@@ -324,4 +324,23 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       FsConstants.LOCAL_FS_URI, workingDir, new Path("kv1.txt"))
     assert(r === new Path(s"$workingDir/kv1.txt"))
   }
+
+  test("SPARK-37266: Optimize the analysis for view text of persistent view and" +
+    " fix security vulnerabilities caused by sql tampering") {
+    sql("CREATE VIEW parquet_view2 as select * from parquet_tab4")
+    val table = hiveContext.sessionState.catalog.getTableMetadata(TableIdentifier("parquet_view2"))
+    try {
+      // Simulate the behavior of hackers
+      val tamperedViewText = "drop view parquet_view2"
+      val tamperedTable = table.copy(viewText = Some(tamperedViewText))
+      hiveContext.sessionState.catalog.alterTable(tamperedTable)
```

Comment:
Seems like we can run this test in sql/core, as it does not rely on hive. Can we move it to `PersistedViewTestSuite`?